src/spyglass/lfp/analysis/v1/lfp_band.py

```diff
@@ -109,57 +137,133 @@ def set_lfp_band_electrodes(
+            ref_list_final = [reference_electrode_list] * len(electrode_list)
+        elif isinstance(reference_electrode_list, (list, np.ndarray)):
+            temp_ref_list = list(reference_electrode_list)
+            if len(temp_ref_list) == 1:
+                ref_list_final = [temp_ref_list[0]] * len(
+                    electrode_list
+                )  # Broadcast list of 1
+            elif len(temp_ref_list) == len(electrode_list):
+                ref_list_final = temp_ref_list
+            else:
+                raise ValueError(
+                    "reference_electrode_list (if list/array) must contain either 1 or len(electrode_list) elements"
+                )
+        else:
+            raise TypeError(
+                "reference_electrode_list must be an int, list, or numpy array."
             )
-        # add a -1 element to the list to allow for the no reference option
-        available_electrodes = np.append(available_electrodes, [-1])
-        if not np.all(np.isin(reference_electrode_list, available_electrodes)):
+
+        # Now validate the contents of ref_list_final
+        available_electrodes_check = np.append(
+            available_electrodes.copy(), [-1]
+        )  # Use original available_electrodes
+        if not np.all(np.isin(ref_list_final, available_electrodes_check)):
             raise ValueError(
-                "All elements in reference_electrode_list must be valid "
-                "electrode_ids in the LFPSelection table"
+                "All elements in reference_electrode_list must be valid electrode_ids or -1."
             )
 
-        # make a list of all the references
-        ref_list = np.zeros((len(electrode_list),))
-        ref_list[:] = reference_electrode_list
+        # Ensure ref_list is a numpy array of int for zipping later
+        ref_list = np.array(ref_list_final, dtype=int)
+
+        # Sort electrode_list and ref_list together
+        # This ensures that the order of electrodes and references is consistent
+        electrode_sort_ind = np.argsort(np.array(electrode_list, dtype=int))
+        electrode_list = np.array(electrode_list, dtype=int)[electrode_sort_ind]
+        ref_list = ref_list[electrode_sort_ind]
+
+        if lfp_band_sampling_rate is None:
+            # if no sampling rate is provided, use the default
+            lfp_band_sampling_rate = lfp_sampling_rate
+        else:
+            logger.info(
+                "It is recommended to use the same sampling rate as the original "
+                + "lfp data to avoid aliasing."
+            )
+        if lfp_band_sampling_rate > lfp_sampling_rate:
+            raise ValueError(
+                "lfp_band_sampling_rate must be less than or equal to the "
+                + "lfp sampling rate"
+            )
+        if lfp_band_sampling_rate <= 0:
+            raise ValueError("lfp_band_sampling_rate must be greater than 0")
+        decimation = lfp_sampling_rate // lfp_band_sampling_rate
```

Comment:
`common_ephys` and `lfp.py` have some of the same sampling rate logic. Is it worth extracting this input validation to a util that the others can benefit from?